### PR TITLE
chore: various renames and fixes

### PR DESCRIPTION
**Pull request by Jmgr**
## Changes

* removes a few class attributes definition (they should be instance attributes instead)
* renames `inner` to `child` (`producer` was also considered, but `child` is more idiomatic in the context of an AST)
* renames `inner_type` to `contained_type` in compound types (we are now storing more than one type in compound types) and removed `inner_type` from operations
* renames `to_type` to `to_mir`
* adds `is_scalar` and `is_literal` (will be used in a follow up PR)

Note that this doesn't break the MIR. We will do the necessary changes in a follow up PR, as well as in another PR in the main nillion repo so that it is able to interpret the changes.

You can review each commit separately instead of all changes at once if you want, should be easier to read.

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/NillionNetwork/nada-dsl/blob/main/CONTRIBUTING.md) guidelines followed
* [x] Unit tests added/updated (if applicable)
* [ ] Backwards compatibility analysis completed (if applicable). "Will this change require recompilation and upload of user programs?"

### CODE REVIEW

File: nada_dsl/ast_util.py [hunk 1]
```diff
@@ -88,17 +88,17 @@ class UnaryASTOperation(ASTOperation):
     """Superclass of all the unary operations in AST representation"""
 
     name: str
-    inner: int
+    this: int
```
Comment (lumasepa) [inline nada_dsl/ast_util.py:91]:
```suggestion
    child: int
```

File: nada_dsl/ast_util.py [hunk 2]
```diff
@@ -88,17 +88,17 @@ class UnaryASTOperation(ASTOperation):
...
 
-    def inner_operations(self):
-        return [self.inner]
+    def child_operations(self):
+        return [self.this]
```
Comment (lumasepa) [inline nada_dsl/ast_util.py:94]:
```suggestion
        return [self.child]
```

File: nada_dsl/ast_util.py [hunk 3]
```diff
@@ -88,17 +88,17 @@ class UnaryASTOperation(ASTOperation):
...
 
     def to_mir(self):
 
         return {
             self.name: {
                 "id": self.id,
-                "this": self.inner,
+                "this": self.this,
```
Comment (lumasepa) [inline nada_dsl/ast_util.py:101]:
```suggestion
                "child": self.child,
```

File: nada_dsl/ast_util.py [hunk 4]
```diff
@@ -113,7 +113,7 @@ class IfElseASTOperation(ASTOperation):
     arg_0: int
     arg_1: int
 
-    def inner_operations(self):
+    def child_operations(self):
         return [self.this, self.arg_0, self.arg_1]
```
Comment (lumasepa) [inline nada_dsl/ast_util.py:117]: I will rename `this` to `condition`, even being peaky, `condition_child`, `true_branch_child`, `false_branch_child`

**Review by lumasepa — APPROVED** [review]

**Review by pablojhl — APPROVED** [review]

**Comment by Jmgr** [comment]
Sorry I had to make another change to this PR. As we discussed in Slack making breaking changes to the MIR now is not ideal, and we will also need to rename `inner_type` in a lot of places, so for now we are going to keep internal changes only.

**Review by navasvarela — APPROVED** [review]

File: nada_mir/src/nada_mir_proto/nillion/nada/types/v1/__init__.py [hunk 5]
```diff
@@ -48,7 +48,7 @@ class Object(betterproto.Message):
 class Array(betterproto.Message):
     """Array type, defines a collection of homogeneous values"""
 
-    inner_type: "NadaType" = betterproto.message_field(1)
+    contained_types: "NadaType" = betterproto.message_field(1)
```
[the remark this answers is not in the document]
Comment (Jmgr) [reply nada_mir/src/nada_mir_proto/nillion/nada/types/v1/__init__.py:51]: What is the process to update those files? The corresponding .proto files are not in this repo, but in https://github.com/NillionNetwork/nada-mir-model -- they are not used now AFAIK, but is the plan in the future to add there here?

[the remark this answers is not in the document]
Comment (navasvarela) [reply nada_mir/src/nada_mir_proto/nillion/nada/types/v1/__init__.py]: There's a script under `nada_mir/scripts/gen_proto.sh`. It generates the code, using a local folder where the .proto are.

**Pull request merged**